test/e2e/framework/networking_utils.go

```diff
@@ -186,13 +186,12 @@ func (config *NetworkingTestConfig) DialFromContainer(protocol, containerIP, tar
 			if err := json.Unmarshal([]byte(stdout), &output); err != nil {
 				Logf("WARNING: Failed to unmarshal curl response. Cmd %v run in %v, output: %s, err: %v",
 					cmd, config.HostTestContainerPod.Name, stdout, err)
-				continue
-			}
-
-			for _, hostName := range output["responses"] {
-				trimmed := strings.TrimSpace(hostName)
-				if trimmed != "" {
-					eps.Insert(trimmed)
+			} else {
+				for _, hostName := range output["responses"] {
+					trimmed := strings.TrimSpace(hostName)
+					if trimmed != "" {
+						eps.Insert(trimmed)
+					}
```

Comment:
there's another case where we call this function to make sure no endpoints are returned in say 100 tries. We do this after we switch the service from nodePort -> normal clusterIP, for example. We give it an empty list of endpoints, and make sure that 0 endpoitns are returned in 100 iterations of this loop. 

With your change, won't that case pass even if we keep hitting errors? 
I'm specifically asking about always hitting the json.Unmarshal error in this call: https://github.com/kubernetes/kubernetes/blob/master/test/e2e/networking.go#L174
That should fail, can you confirm?